crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -256,6 +262,36 @@ async fn volume_fee_buy_order_test(web3: Web3) {
     .await;
 }
 
+async fn price_improvement_fee_sell_order_test(web3: Web3) {
+    let fee_policy = FeePolicyKind::PriceImprovement {
+        factor: 0.3,
+        max_volume_factor: 0.9,
+    };
+    // Without protocol fee:
+    // Expected execution is 10000000000000000000 GNO for
+    // 9871415430342266811 DAI, with executed_surplus_fee = 167058994203399 GNO
+    //
+    // protocol fee = 37762575522225 DAI =
+    // 37762575522225 DAI / 9871415430342266811 *
+    // (10000000000000000000 - 167058994203399) = 38253829889851 GNO
+    //
+    // final execution is 10000000000000000000 GNO for 9871377667766744586 DAI, with
+    // executed_surplus_fee = 205312824093250 GNO
+    //
+    // Settlement contract balance after execution = 205312824093250 GNO =
+    // 205312824093250 GNO * 9871377667766744586 / (10000000000000000000 -
+    // 205312824093250) = 202676203868401 DAI
+    execute_test(
```

Comment:
Is this `execute_test` wrapper really necessary assuming we get rid of all the redundant surplus test cases and just keep 3 cases (surplus, price improvement and volume fee)? I think having three more verbose test cases would be better.

Maybe we could use more meaningful values (currently it seems like these were derived by running the test once and adjusting the values to what the test created) so it might not be that useful. But then, I also don't want us to go down the rabbit hole of redesigning the whole test infra like we did in the driver crate. Is it possible to use limit orders to ensure predictable match amounts?

Otherwise we can maybe do something like
1. Seed a bit of liquidity on the AMM that serves the token pair
2. Get a quote (which we use for placing the order)
3. Inject more liquidity into the AMM (this should improve the price)
4. Get another quote which we can do to do the math of how much the user should get in the end of the day.

This way it would be a bit easier to follow that these long digit series.

Reply:
Will be tackled in the https://github.com/cowprotocol/services/issues/2468